mpv.py

```diff
@@ -880,6 +897,12 @@ def _loop(self):
                     if target in self._message_handlers:
                         self._message_handlers[target](*args)
 
+                if eid == MpvEventID.COMMAND_REPLY:
+                    key = devent['reply_userdata']
+                    callback = self._event_async_callbacks.pop(key, None)
+                    if callback:
+                        callback(devent['error'], devent['event']['result'])
```

Comment:
I think if we support ```None``` callbacks with the special ```2**64-1``` userdata key below we should provide a default error handler here: If callback is None, but an error was returned, I think we should ```warnings.warn()``` here to make mistakes more obvious.

Reply:
If we use the special `MPV._UINT_MAX = 2 ** 64 - 1` userdata key for `None` callbacks, we won't even store the `None` callback in the dictionary (original `_register_async_callback` implementation, see snippet in above discussion). So if the pop operation returns a callback set to `None`, this would be an error in itself.

We could of course still save `None` callbacks in the `_register_async_callback` function in the dictionary. Then the pop operation may pop `None` callbacks and we would be able to inform the user.

As the later sounds better from a user perspective I guess we should go with that :+1:

Reply:
But again we might override some callbacks functions when the user does not specify a callback (set to `None`).
Hm, so maybe it would be better if we drop the special treatment of `None` callbacks.
But then we would need to lock on every async command ... 
Which solution do you prefer?

Reply:
Done in latest commit.

Mhm, I wonder if we should save the command in the `self._event_async_callbacks` dictionary, too.
So instead of the callback we might store a tuple(command, callback), what do you think?
The user probably wants to know, which command failed. And the way it's implemented right know doesn't provide the information.

Reply:
I think the nice thing about using the constant is that it avoids storing anything on ongoing fire-and-forget async calls. If we accept a few bytes per call in the dict, we can just have the callback default to one that raises mpv errors as warnings, something like this:
```def command_async(name, *args, callback=None):
    if callback is None:
        def callback(err, _result):
            warnings.warn('Error executing async {} {} command: {}'.format(name, ' '.join(repr(arg) for arg in args), err))

This would be the simplest, most readable solution. Perhaps we should do this, and then only introduce optimizations like the special "no callback" key once we notice they're necessary?

Reply:
I've added the default error handler (latest commit) to the code that stores the callback in the dictionary.

> This would be the simplest, most readable solution. Perhaps we should do this, and then only introduce optimizations like the special "no callback" key once we notice they're necessary?

Sounds good to me :smiley: